
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>Blog - Reason I Am Here - Nacho Caballero</title>
    
    <meta name="author" content="Nacho Caballero">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <!-- Le HTML5 shim, for IE6-8 support of HTML elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Le styles -->
    <link href="/assets/themes/twitter/css/mailchimp.css" rel="stylesheet" type="text/css" media="all">
    <link href="/assets/themes/twitter/bootstrap/css/bootstrap.2.2.2.min.css" rel="stylesheet">
    <link href="/assets/themes/twitter/css/style.css?body=1" rel="stylesheet" type="text/css" media="all">

    <!--highlight-->
    <link rel="stylesheet" href="/assets/themes/twitter/highlight.js/styles/tomorrow-night-eighties.css">

    <link href="/assets/themes/twitter/css/nacho.css?body=1" rel="stylesheet" type="text/css" media="all">
    <script src="/assets/themes/twitter/highlight.js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>


    <script src="/assets/themes/twitter/jquery/jquery-1.11.1.min.js"></script>
    <script src="/assets/themes/twitter/jquery/jquery.lazyload.min.js"></script>
    <script type="text/javascript" charset="utf-8">
        $(function() {
          $("img.lazy").show().lazyload({
            effect: "fadeIn",
            threshold : 200
          });
        });
    </script>

    <!-- Le fav and touch icons -->
  <!-- Update these with your own images
    <link rel="shortcut icon" href="images/favicon.ico">
    <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
  -->

    <!-- atom & rss feed -->
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">

  </head>

  <body>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container-narrow">
          <a class="brand" href="/">Reason I am here</a>
          <ul class="nav">
            
            
            


  
    
      
      	
      	<li><a href="/about.html">Nacho Caballero</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="/archive.html">Article Index</a></li>
      	
      
    
  
    
      
    
  
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	<li><a href="/tags.html">Tags</a></li>
      	
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
  
    
  



          </ul>
        </div>
      </div>
    </div>

    <div class="container-narrow">

      <div class="content">
        


<div class="row-fluid">
  <div class="span12">
    <div id="home">
  <ul class="posts">
    
    <h1><a href="/2013/09/26/switching-between-long-and-wide-formats-in-r">Switching Between Long and Wide Formats in R</a></h1>
    <p class="author">
        <span class="date">26 September 2013</span>
    </p>
    <div class="content">
        <p>This is a basic tutorial in R for switching between the two most common data formats: wide and long. The example dataset we will use is made up of <a href="https://en.wikipedia.org/wiki/Rt_pcr">RT-PCR</a> threshold cycle (Ct) values under two conditions (A and B), with two replicates each.</p>


    </div>
    <hr>
    
    <h1><a href="/2013/09/25/bioinformatics-at-the-bench">Bioinformatics at the Bench</a></h1>
    <p class="author">
        <span class="date">25 September 2013</span>
    </p>
    <div class="content">
        <p>Most bioinformaticians don&rsquo;t like to do experiments. We love the biological stories, but give us a pipette and we will quickly groan. I have spent the past two years in a virology lab, mainly working on computational problems, and eternally debating if I should plunge into the pool of experimental biology. Up until now, I have only dipped my toes in a few basic experiments, but I have recently decided to stop working on purely computational projects and transition to ones that require experimental skills. Here are my reasons.</p>


    </div>
    <hr>
    
    <h1><a href="/2013/09/24/why-10-of-the-population-hates-cilantro-and-the-rest-doesnt-know-any-better">Why 10% of the Population Hates Cilantro and the Rest Doesn't Know Any Better</a></h1>
    <p class="author">
        <span class="date">24 September 2013</span>
    </p>
    <div class="content">
        <blockquote>
<pre>
"No cilantro, please"
Yet I know when it arrives
It will be present</pre>

<p>Haiku by <a href="http://www.ihatecilantro.com/member.php?user=Krakrs">Krakrs</a>, a member of <a href="http://ihatecilantro.com">ihatecilantro.com</a></p>
</blockquote>

<p>The first time I tried cilantro I didn&rsquo;t realize it; I just thought somebody had emptied a bottle of Old Spice on my pizza in an attempt to poison me. Cilantro tastes like soap to approximately 10% of the people who have had their genotype analyzed by <a href="http://www.biomedcentral.com/content/pdf/2044-7248-1-22.pdf">23andMe</a>. The currently accepted explanation is that those of us who passionately despise cilantro were born with a genetic variant known as a <strong>single-nucleotide polymorphism</strong> (or <a href="http://en.wikipedia.org/wiki/Single-nucleotide_polymorphism">SNP</a>, pronounced &lsquo;snip&rsquo;).</p>


    </div>
    <hr>
    
    <h1><a href="/2013/09/23/7-r-tips-rescued-from-the-vault">7 R Tips Rescued from the Vault</a></h1>
    <p class="author">
        <span class="date">23 September 2013</span>
    </p>
    <div class="content">
        <p>I keep a 2000-lines long <a href="https://github.com/nachocab/tips_and_tricks/blob/master/R_tricks.R">file</a> full of R commands that I considered worthy of remembering at some point during the past 5 years. Life is too short, so here are 7 tips that don&rsquo;t get enough publicity, ranked from most to least awesome.</p>


    </div>
    <hr>
    
    <h1><a href="/2013/09/22/how-to-stop-sabotaging-your-work">How to Stop Sabotaging Your Work</a></h1>
    <p class="author">
        <span class="date">22 September 2013</span>
    </p>
    <div class="content">
        <p>I keep forgetting how to defend myself against the feeling of <strong>being stuck</strong>—writers call it writer&rsquo;s block, painters call it feeling uninspired, runners don&rsquo;t have a name for it, they just suck it up and keep running—, so this post is a reminder on the two techniques that help me neutralize the resistance to get back to work.</p>

<p><img src="https://lh5.googleusercontent.com/NhThbkn4hN5h_kj53hvrsVapMzrdqXK2zAxFLhirMnU=w561-h824-no" alt="basket"></p>


    </div>
    <hr>
    
    <h1><a href="/2013/09/21/measuring-the-effects-of-interferon-stimulated-genes-on-viral-replication">Measuring the Effects of Interferon Stimulated Genes on Viral Replication</a></h1>
    <p class="author">
        <span class="date">21 September 2013</span>
    </p>
    <div class="content">
        <p>Whenever I read a paper that describes an experimental technique that I&rsquo;m not familiar with (which is often, since the lab bench is not where bioinformatic students spend most their time), I compensate by doing background research. Lately, I have been reading a lot about the antiviral effects of interferon-stimulated genes (ISGs), and since <a href="http://www.utsouthwestern.edu/labs/schoggins/">John Schoggins</a> is giving a seminar in my university on Monday, I thought I would post about a clever technique that he used in his <a href="http://www.nature.com/nature/journal/v472/n7344/abs/nature09907.html">Nature paper</a> to quantify the impact of ISGs on viral replication.</p>


    </div>
    <hr>
    
    <h1><a href="/2013/09/20/my-phd-research-project-diagnosing-viral-infections">My PhD Research Project: Diagnosing Viral Infections</a></h1>
    <p class="author">
        <span class="date">20 September 2013</span>
    </p>
    <div class="content">
        <p>I gave a presentation today in the Microbiology departmental retreat at Boston University, so I though I&rsquo;d share it.</p>

<p><img src="https://lh5.googleusercontent.com/-nzVbaorZHoY/UjzzWilVUuI/AAAAAAAAEUA/5--h0EgJKbU/w500-no/lassa+title.png" alt="presentation"></p>

<p>The goal of the research project I have been working on for the past two years, as part of my Bioinformatics PhD at Boston University, is to use blood samples from infected patients to determine which virus is causing the infection.</p>

<p>Traditional diagnostic methods, like detection by <a href="https://en.wikipedia.org/wiki/ELISA">ELISA</a>, are only effective after the virus has had enough time to replicate in the blood of the patient. However, for some viral infections, when this happens it is already too late for the treatment to be effective.</p>

<p>The approach that we have taken in my lab is to measure the transcriptional changes that take place in the peripheral blood cells, and identify patterns of expression that are unique for each type of infection. This indirect way of viral detection has the potential to reduce diagnosis time by several days, since it is known that these transcriptional changes precede the appearance of virus particles in the blood, and we believe that they are specific enough to discriminate between viruses.</p>


    </div>
    <hr>
    
    <h1><a href="/2013/09/19/working-with-d3-visualizations-in-adobe-illustrator">Working with D3 Visualizations in Adobe Illustrator</a></h1>
    <p class="author">
        <span class="date">19 September 2013</span>
    </p>
    <div class="content">
        <p><a href="http://d3js.org/">D3</a> is an awesome tool to build interactive visualizations. I use it extensively for the initial stages of my research projects, but converting D3 charts from SVG to PDF and polishing them for publication can be a harrowing experience. This post describes a few techniques that make this process more pleasant. If you have any additional ideas, please leave a comment at the bottom of the post.</p>


    </div>
    <hr>
    
    <h1><a href="/2013/09/18/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-3">AWK GTF! How to Analyze a Transcriptome Like a Pro - Part 3</a></h1>
    <p class="author">
        <span class="date">18 September 2013</span>
    </p>
    <div class="content">
        <p>This post belongs to a 3-part series on AWK: <a href="http://reasoniamhere.com/2013/09/16/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-1">Part 1</a> - <a href="http://reasoniamhere.com/2013/09/17/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-2">Part 2</a> - <strong>Part 3</strong></p>

<p>We finished <a href="http://reasoniamhere.com/2013/09/17/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-2">Part 2</a> by creating an associative array, increasing the value of each of its keys for every line that contained a matching key, and used the <code>END</code> rule to print each key&rsquo;s final count.</p>


    </div>
    <hr>
    
    <h1><a href="/2013/09/17/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-2">AWK GTF! How to Analyze a Transcriptome Like a Pro - Part 2</a></h1>
    <p class="author">
        <span class="date">17 September 2013</span>
    </p>
    <div class="content">
        <p>This post belongs to a 3-part series on AWK: <a href="http://reasoniamhere.com/2013/09/16/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-1">Part 1</a> - <strong>Part 2</strong> - <a href="http://reasoniamhere.com/2013/09/18/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-3">Part 3</a></p>

<p>Back in <a href="http://reasoniamhere.com/2013/09/16/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-1">Part 1</a>, we learned how to tell AWK to select specific lines from a tab-separated <a href="http://reasoniamhere.com/2013/09/16/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-1/#get_the_data">transcriptome file</a> (using the rule <code>&#39;$3 == &quot;gene&quot;&#39;</code>) and return a specific column (using the <code>print</code> statement inside the curly braces <code>{ }</code>).</p>

<p>In this post, we are going to count how many exons make up each protein-coding gene.</p>


    </div>
    <hr>
    

    <div id="post-pagination" class="pagination">
      
        <p class="previous">
          
            <a href="/page3">« Previous</a>
          
        </p>
      

      
        <p class="next">
          <a href="/page5">Next »</a>
        </p>
      

    </div>
  </ul>
</div>





  </div>
</div>


      </div>

      <footer>
        <p class="copyright">&copy; 2015 Nacho Caballero</p>
        <p>
          <a href="http://feeds.feedburner.com/TheReasonIAmHere">RSS</a> •
          <a href="https://twitter.com/nachocaballero">Twitter</a> •
          <a href="http://reasoniamhere.com/about.html">About</a>
        </p>
      </footer>

    </div>
    <script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-42664204-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>
  </body>
</html>

