
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>AWK GTF! How to Analyze a Transcriptome Like a Pro - Part 3 - Blog - Reason I Am Here - Nacho Caballero</title>
    
    <meta name="author" content="Nacho Caballero">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <!-- Le HTML5 shim, for IE6-8 support of HTML elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Le styles -->
    <link href="/assets/themes/twitter/css/mailchimp.css" rel="stylesheet" type="text/css" media="all">
    <link href="/assets/themes/twitter/bootstrap/css/bootstrap.2.2.2.min.css" rel="stylesheet">
    <link href="/assets/themes/twitter/css/style.css?body=1" rel="stylesheet" type="text/css" media="all">

    <!--highlight-->
    <link rel="stylesheet" href="/assets/themes/twitter/highlight.js/styles/tomorrow-night-eighties.css">

    <link href="/assets/themes/twitter/css/nacho.css?body=1" rel="stylesheet" type="text/css" media="all">
    <script src="/assets/themes/twitter/highlight.js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>


    <script src="/assets/themes/twitter/jquery/jquery-1.11.1.min.js"></script>
    <script src="/assets/themes/twitter/jquery/jquery.lazyload.min.js"></script>
    <script type="text/javascript" charset="utf-8">
        $(function() {
          $("img.lazy").show().lazyload({
            effect: "fadeIn",
            threshold : 200
          });
        });
    </script>

    <!-- Le fav and touch icons -->
  <!-- Update these with your own images
    <link rel="shortcut icon" href="images/favicon.ico">
    <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
  -->

    <!-- atom & rss feed -->
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">

  </head>

  <body>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container-narrow">
          <a class="brand" href="/">Reason I am here</a>
          <ul class="nav">
            
            
            


  
    
      
      	
      	<li><a href="/about.html">Nacho Caballero</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="/archive.html">Article Index</a></li>
      	
      
    
  
    
      
    
  
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	<li><a href="/tags.html">Tags</a></li>
      	
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
  
    
  



          </ul>
        </div>
      </div>
    </div>

    <div class="container-narrow">

      <div class="content">
        

<div class="page-header">
  <h1>AWK GTF! How to Analyze a Transcriptome Like a Pro - Part 3 </h1>
</div>

<div class="row-fluid post-full">
  <div class="span12">
      <span class="date">18 September 2013</span>
    <div class="content">
      <p>This post belongs to a 3-part series on AWK: <a href="http://reasoniamhere.com/2013/09/16/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-1">Part 1</a> - <a href="http://reasoniamhere.com/2013/09/17/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-2">Part 2</a> - <strong>Part 3</strong></p>

<p>We finished <a href="http://reasoniamhere.com/2013/09/17/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-2">Part 2</a> by creating an associative array, increasing the value of each of its keys for every line that contained a matching key, and used the <code>END</code> rule to print each key&rsquo;s final count.</p>

<!--excerpt-->

<p>In this post, we will apply the translator pattern to turn the gene symbols used in the <a href="http://reasoniamhere.com/2013/09/16/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-1/#get_the_data">transcriptome</a> we have been working with into official <a href="http://www.genenames.org/cgi-bin/hgnc_stats">HGNC symbols</a>.</p>

<h2>Calling AWK with an external file</h2>

<p>We have been calling AWK directly from the command line, but this approach can be limiting sometimes. The alternative is to use the <code>-f</code> option to tell AWK where the code is (in this case, in the <code>source_counter.awk</code> file).</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="c"># create the instructions for AWK</span>
<span class="nb">echo</span> <span class="s1">&#39;NR &gt; 5 {</span>
<span class="s1">    source_counter[$2] += 1</span>
<span class="s1">}</span>

<span class="s1">END {</span>
<span class="s1">    for (source_name in source_counter){</span>
<span class="s1">        print source_name, source_counter[source_name]</span>
<span class="s1">    }</span>
<span class="s1">}&#39;</span> &gt; source_counter.awk

<span class="c"># run AWK</span>
awk -f source_counter.awk transcriptome.gtf

HAVANA 2257550
ENSEMBL 357012
</code></pre></div>
<p><code>NR</code> is a special variable that contains the current line number (AWK calls lines records, which explains the R). <code>NR &gt; 5</code> tells AWK to execute the code between the curly braces only when the line number is greater than 5, so it skips the first 5 lines. We do this to get rid of the header lines that appear at the beginning of the Gencode transcriptome file.</p>

<p>Remember you still have to provide AWK with an input file (<code>transcriptome.gtf</code>) or use a pipe (<code>cat transcriptome.gtf | awk -f source_counter.awk</code>). If you don&rsquo;t, AWK will just sit there quietly until it receives an input that will never arrive. This is bound to happen to you at some point; when it does, press <code>ctrl c</code>, smack your forehead and start again.</p>

<h2>Splitting strings</h2>

<p>The following two commands create the files that we will use for the rest of the post. Run them so you can play along.</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nb">echo</span> <span class="s1">&#39;</span>
<span class="s1">ZYX_CTSL 40</span>
<span class="s1">WVU_CTSV 21</span>
<span class="s1">TSR_MYCL 11</span>
<span class="s1">QPO_KMT2A 155</span>
<span class="s1">NML_HSP90AB1 36</span>
<span class="s1">KJI_CAPN7 52</span>
<span class="s1">HGF_DACH2 84</span>
<span class="s1">EDC_IFNA14 1&#39;</span> &gt; exon_counts.txt

<span class="c"># the format of the translator is UNOFFICIAL_GENE_NAME OFFICIAL_GENE_NAME</span>
<span class="nb">echo</span> <span class="s1">&#39;</span>
<span class="s1">CTSL CTSL1</span>
<span class="s1">FLJ31037 CTSL1</span>
<span class="s1">MGC125957 CTSL2</span>
<span class="s1">CTSU CTSL2</span>
<span class="s1">CTSV CTSL2</span>
<span class="s1">CATL2 CTSL2</span>
<span class="s1">MYCL MYCL1</span>
<span class="s1">LMYC MYCL1</span>
<span class="s1">KMT2A MLL</span>
<span class="s1">ALL-1 MLL</span>
<span class="s1">HTRX1 MLL</span>
<span class="s1">KMT2A MLL</span>
<span class="s1">CXXC7 MLL</span>
<span class="s1">TET1-MLL MLL</span>
<span class="s1">MLL1A MLL</span>
<span class="s1">FLJ11783 MLL&#39;</span> &gt; translator.txt
</code></pre></div>
<p>Imagine that you found <code>exon_counts.txt</code> in the supplementary materials section of a paper from a rival lab, and you want to compare their results with yours. Unfortunately, they seem to enjoy prepending their gene symbols with a strange (but vaguely familiar) code followed by an underscore. How do we get rid of it?</p>

<p>AWK to the rescue. We can use <code>split</code>, one of its <a href="http://www.gnu.org/software/gawk/manual/html_node/String-Functions.html">functions</a> to manipulate strings:</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nb">echo</span> <span class="s2">&quot;HOLY-GUACAMOLE&quot;</span> <span class="p">|</span> awk <span class="s1">&#39;{ split($1, words, &quot;-&quot;); print words[2] }&#39;</span> <span class="c"># GUACAMOLE</span>
<span class="nb">echo</span> <span class="s2">&quot;HOLY-GUACAMOLE&quot;</span> <span class="p">|</span> awk <span class="s1">&#39;{ split($1, words, &quot;-&quot;); print words[1] }&#39;</span> <span class="c"># HOLY</span>
<span class="nb">echo</span> <span class="s2">&quot;HOLY-GUACAMOLE-CHIMICHANGA&quot;</span> <span class="p">|</span> awk <span class="s1">&#39;</span>
<span class="s1">{</span>
<span class="s1">    split($1, words, &quot;-&quot;)</span>
<span class="s1">    print &quot;I would like&quot;, words[2], &quot;in my&quot;, words[3] &quot;, please&quot;</span>
<span class="s1">}&#39;</span> <span class="c"># I would like GUACAMOLE in my CHIMICHANGA, please</span>
</code></pre></div>
<p>Its first argument is the <strong>string</strong> that we want to split, the second is the name of the <strong>array</strong> variable that will contain the pieces of the string that we split, and the third is the <strong>delimiter</strong> that we want to use to separate the string into pieces. In this case, we chose to call the array variable <code>words</code>. An array can be considered as a special case of the associative array, where the keys are numbers, instead of words.</p>

<p>The semicolon in the first two commands is just a way to be lazy and write more than one function (<code>split</code>, <code>print</code>) in the same line.</p>

<p>Notice a small subtlety hidden among the Tex-Mex: there is no comma between <code>words[3]</code> and <code>&quot;, please&quot;</code>. If there was, the output would be <code>CHIMICHANGA , please</code>. Because there is no comma, AWK simply appends the two strings without inserting the default output separator (the space).</p>

<p>Now we are ready to get rid of the reverse-alphabetical madness.</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">awk <span class="s1">&#39;{</span>
<span class="s1">    split($1, words, &quot;_&quot;)</span>
<span class="s1">    print words[2], $2</span>
<span class="s1">}&#39;</span> exon_counts.txt &gt; exon_counts2.txt

cat exon_counts2.txt

CTSL 40
CTSV 21
MYCL 11
KMT2A 155
HSP90AB1 36
CAPN7 52
DACH2 84
IFNA14 1
</code></pre></div>
<h2>Building a translation dictionary</h2>

<p>Watch the magic unfold:</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">awk -v <span class="nv">translations_file</span><span class="o">=</span><span class="s2">&quot;translator.txt&quot;</span> <span class="s1">&#39;</span>
<span class="s1">BEGIN{</span>
<span class="s1">    while (getline &lt; translations_file) {</span>
<span class="s1">        translations[$1] = $2</span>
<span class="s1">    }</span>
<span class="s1">    close(translations_file)</span>
<span class="s1">}</span>

<span class="s1">{</span>
<span class="s1">    if (translations[$1] != &quot;&quot;){</span>
<span class="s1">       $1 = translations[$1]</span>
<span class="s1">       $3 = &quot;translated&quot;</span>
<span class="s1">    } else {</span>
<span class="s1">       $3 = &quot;untranslated&quot;</span>
<span class="s1">    }</span>
<span class="s1">   print</span>
<span class="s1">}&#39;</span> exon_counts2.txt

<span class="c"># output</span>
CTSL1 <span class="m">40</span> translated
CTSL2 <span class="m">21</span> translated
MYCL1 <span class="m">11</span> translated
MLL <span class="m">155</span> translated
HSP90AB1 <span class="m">36</span> untranslated
CAPN7 <span class="m">52</span> untranslated
DACH2 <span class="m">84</span> untranslated
IFNA14 <span class="m">1</span> untranslated
</code></pre></div>
<p>I always have to look up the exact syntax for the translator pattern because I never remember, but it&rsquo;s so useful that I keep it in an easily accessible <a href="https://github.com/nachocab/tips_and_tricks/blob/master/bash_tricks.sh">text file</a>, along with all the other useful-but-impossible-to-remember commands.</p>

<p>A lot is going on here.</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">awk -v <span class="nv">translations_file</span><span class="o">=</span><span class="s2">&quot;translator.txt&quot;</span> <span class="s1">&#39;...&#39;</span> exon_counts2.txt
</code></pre></div>
<p>The <code>-v</code> option is used to create variables that can be used by AWK in the code that goes between the single quotes. In this case we are creating a variable called <code>translations_file</code> with value <code>&quot;translator.txt&quot;</code>. Due to the Holy UNIX Commandments, no spaces are allowed around the equals sign (this rule doesn&rsquo;t apply inside AWK, though).</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">BEGIN<span class="o">{</span> ... <span class="o">}</span>
</code></pre></div>
<p>The <code>BEGIN</code> rule does the opposite of the <code>END</code> rule: it is true before the input file (<code>exon_counts2.txt</code>) is read, and false thereafter.</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">while</span> <span class="o">(</span>condition<span class="o">)</span> <span class="o">{</span> ... <span class="o">}</span>
</code></pre></div>
<p>Similar to the <code>for</code> loop that we saw in the previous post, there is also a <code>while</code> loop, which runs until the condition between the parentheses stops being true.</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">while</span> <span class="o">(</span>getline &lt; translations_file<span class="o">)</span> <span class="o">{</span>
    translations<span class="o">[</span><span class="nv">$1</span><span class="o">]</span> <span class="o">=</span> <span class="nv">$2</span>
<span class="o">}</span>
close<span class="o">(</span>translations_file<span class="o">)</span>
</code></pre></div>
<p><code>getline &lt; some_file</code> is an <a href="http://www.gnu.org/software/gawk/manual/html_node/Getline.html#Getline">internal</a> AWK command that lets you read the next line from a different file than the standard input file. After the line is read, the dollar <code>$</code> variables are updated, so we can use <code>$1</code> and <code>$2</code> just like we usually do. The file must then be closed to avoid angering the UNIX gods.</p>

<p><em>NOTE</em>: There is a lot of wizardry that surrounds the <code>getline</code> command. Ninety percent of the time, the translator pattern is enough. If you want to do fancier stuff, like using a translator file with a different input delimiter than the one used by the input file, it&rsquo;s worth spending an hour reading the <a href="http://www.gnu.org/software/gawk/manual/html_node/Getline.html#Getline">documentation</a> to understand all the nitty-gritties.</p>

<p><code>translations</code> is an associative array that has the bad gene symbols as keys, and the official gene symbols as values. The <code>translator.txt</code> file doesn&rsquo;t have any repeats in column one, so we can be sure that each bad symbol only has one good translation. But our toy translator file doesn&rsquo;t have all the gene symbols that appear in <code>exon_counts2.txt</code> (HSP90AB1 is missing, for example).</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">if</span> <span class="o">(</span>translations<span class="o">[</span><span class="nv">$1</span><span class="o">]</span> !<span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="o">){</span>
   <span class="nv">$1</span> <span class="o">=</span> translations<span class="o">[</span><span class="nv">$1</span><span class="o">]</span>
   <span class="nv">$3</span> <span class="o">=</span> <span class="s2">&quot;translated&quot;</span>
<span class="o">}</span> <span class="k">else</span> <span class="o">{</span>
   <span class="nv">$3</span> <span class="o">=</span> <span class="s2">&quot;untranslated&quot;</span>
<span class="o">}</span>
</code></pre></div>
<p>We check this by using an <code>if</code> statement, which only executes the code inside the braces if what is between the parentheses is true. In our example, <code>translations[$1]</code> will equal the empty string <code>&quot;&quot;</code> for genes genes that don&rsquo;t have a translation. In those cases, we won&rsquo;t overwrite <code>$1</code>.</p>

<p>The <code>print</code> command without any arguments prints all the fields (AWK calls columns <strong>fields</strong>), including the one we created to know if the gene was translated or not.</p>

<p><hr></p>

<p>I hope this series got you interested in learning more about AWK. From personal experience, once you grok it, you want to use it everywhere. There is a megaton of material we didn&rsquo;t cover, but here is a <a href="http://www.catonmat.net/blog/awk-one-liners-explained-part-one/">great link</a> with many handy AWK one-liners. Feel free to get back to me if you have any questions.</p>

    </div>

   <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
   <!-- cilantro -->
   <ins class="adsbygoogle"
        style="display:inline-block;width:300px;height:250px"
        data-ad-client="ca-pub-9441199102473745"
        data-ad-slot="6108086510"></ins>
   <script>
   (adsbygoogle = window.adsbygoogle || []).push({});
   </script>

  

  
    <hr>

    <div class="related">
      <h3>Related posts</h3>
      <ul>
        
          <li><a href="/2013/09/17/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-2">AWK GTF! How to Analyze a Transcriptome Like a Pro - Part 2</a></li>
        
          <li><a href="/2013/09/16/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-1">AWK GTF! How to Analyze a Transcriptome Like a Pro - Part 1</a></li>
        
          <li><a href="/2013/11/14/a-hackers-approach-to-understanding-french-spelling">A Hacker's Approach to Understanding French Spelling</a></li>
        
          <li><a href="/2014/01/11/outrageously-useful-tips-to-master-your-z-shell">Master Your Z Shell with These Outrageously Useful Tips</a></li>
        
          <li><a href="/2013/09/26/switching-between-long-and-wide-formats-in-r">Switching Between Long and Wide Formats in R</a></li>
        
      </ul>
    </div>
  

  <hr>

  
    <ul class="tag_box inline">
      <li><i class="icon-tags"></i></li>
      
      


  
     
    	<li><a href="/tags.html#tutorial-ref">tutorial <span>5</span></a></li>
     
    	<li><a href="/tags.html#bioinformatics-ref">bioinformatics <span>3</span></a></li>
     
    	<li><a href="/tags.html#awk-ref">awk <span>4</span></a></li>
    
  



    </ul>
  

  
  </div>
</div>




      </div>

      <footer>
        <p class="copyright">&copy; 2015 Nacho Caballero</p>
        <p>
          <a href="http://feeds.feedburner.com/TheReasonIAmHere">RSS</a> •
          <a href="https://twitter.com/nachocaballero">Twitter</a> •
          <a href="http://reasoniamhere.com/about.html">About</a>
        </p>
      </footer>

    </div>
    <script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-42664204-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>
  </body>
</html>

