
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>Blog - Reason I Am Here - Nacho Caballero</title>
    
    <meta name="author" content="Nacho Caballero">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <!-- Le HTML5 shim, for IE6-8 support of HTML elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Le styles -->
    <link href="/assets/themes/twitter/css/mailchimp.css" rel="stylesheet" type="text/css" media="all">
    <link href="/assets/themes/twitter/bootstrap/css/bootstrap.2.2.2.min.css" rel="stylesheet">
    <link href="/assets/themes/twitter/css/style.css?body=1" rel="stylesheet" type="text/css" media="all">

    <!--highlight-->
    <link rel="stylesheet" href="/assets/themes/twitter/highlight.js/styles/tomorrow-night-eighties.css">

    <link href="/assets/themes/twitter/css/nacho.css?body=1" rel="stylesheet" type="text/css" media="all">
    <script src="/assets/themes/twitter/highlight.js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>


    <script src="/assets/themes/twitter/jquery/jquery-1.11.1.min.js"></script>
    <script src="/assets/themes/twitter/jquery/jquery.lazyload.min.js"></script>
    <script type="text/javascript" charset="utf-8">
        $(function() {
          $("img.lazy").show().lazyload({
            effect: "fadeIn",
            threshold : 200
          });
        });
    </script>

    <!-- Le fav and touch icons -->
  <!-- Update these with your own images
    <link rel="shortcut icon" href="images/favicon.ico">
    <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
  -->

    <!-- atom & rss feed -->
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">

  </head>

  <body>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container-narrow">
          <a class="brand" href="/">Reason I am here</a>
          <ul class="nav">
            
            
            


  
    
      
      	
      	<li><a href="/about.html">Nacho Caballero</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="/archive.html">Article Index</a></li>
      	
      
    
  
    
      
    
  
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	<li><a href="/tags.html">Tags</a></li>
      	
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
  
    
  



          </ul>
        </div>
      </div>
    </div>

    <div class="container-narrow">

      <div class="content">
        
<div class="page-header">
    <h2>Tag: 'science'</h2>
</div>



    <div class="row">
    <div class="col-lg-12"><h3 class="entry-title"><a href="/2013/10/05/the-truth-about-hair-conditioners-and-blue-eyes">The Truth About Hair Conditioners and Blue Eyes</a></h3></div>
</div>
<div class="row">
    <div class="col-lg-2">
        <div class="post-date">October 05, 2013</div>
    </div>
    <div class="col-lg-8">
        <p><p><img src="https://lh6.googleusercontent.com/-A5o4yzAmY3Y/Uk9qlzh7vqI/AAAAAAAAElY/i8jo5yW8CF0/w500-no/eye-cond.jpg" alt="Eye Conditioner"></p>

<p>I went to get a haircut the other day and my hairdresser convinced me to buy a fancy <a href="https://en.wikipedia.org/wiki/Hair_conditioner#Types">pack conditioner</a>, which, she swore, was a thousand times better than your everyday conditioner, although she didn&rsquo;t know why. After forking out the money, I walked back home fashionably stimulated but scientifically unsatisfied, and I decided to try to decipher the ingredients list from my latest acquisition and understand what made them so special.</p>

<!--excerpt-->

<p>After battling with the unpronounceable names that the International Nomenclature of Cosmetic Ingredients (or INCI) choses for their chemicals, I found that my pack conditioner had many more emulsifying chemicals than a regular conditioner.</p>

<p>An <strong>emulsifier</strong> is a type of surfactant, its molecules encourage the stable mixing of two liquids that wouldn&rsquo;t normally mix. Mayonnaise, for example, is an emulsion of oil and water (egg yolk is 50% water), and lecithin is a fatty component found in egg yolk that acts as the emulsifier. The more emulsifiers you add to a conditioner the more strongly they bind to keratin (the protein that makes up hair). The emulsifying molecules that don&rsquo;t come off after rinsing cover the hair surface and make it look healthier and less dry.</p>

<p>While I was knee-deep researching emulsions, Wikipedia pointed me to the <a href="https://en.wikipedia.org/wiki/Tyndall_effect">Tyndall effect</a>, which explains why shorter-wavelength light is more prone to scattering by small particles than longer-wavelength light; and by extension, why blue eyes don&rsquo;t actually have any blue pigment.</p>

<p>Human irises without melanin are translucentâ€”in effect, an &ldquo;emulsion&rdquo; between a translucent layer and embedded small particles that make it look turbid. When light particles enter the turbid layer, those that have longer wavelengths (which we perceive as red) get through, and the ones with shorter wavelength (which we perceive as blue) get back-scattered by the small particles embedded in the iris. We know that wavelength affects electromagnetic radiation from our everyday experience: radio waves (longer wavelengths) can traverse walls so you can hear NPR, while visible light waves (shorter wavelengths) can be conveniently reflected by simply closing the bathroom door.</p>

<p><img src="https://lh4.googleusercontent.com/-el0S0MrnoW8/Uk9q426KDgI/AAAAAAAAElg/0aOgHH6l50M/w316-h237-no/OPALGLAS2.JPG" alt="glass"></p>

<p>This translucent cryolite glass looks the way an iris does: it shows a different hue depending on the color of the background. From this observation we can infer that implanting a light bulb in the back of the eye (so it wouldn&rsquo;t look dark), would make &ldquo;blue&rdquo; irises look yellow.</p>

<p>Here are some other wonders caused by the Tyndall effect, including <a href="http://www.itp.uni-hannover.de/%7Ezawischa/ITP/scattering.html">why the sky is blue</a>.</p>

<p>I love random science walks.</p>
</p>
    </div>
</div>
<div class="clearfix"></div>
<hr class="list_separator" />

    <div class="row">
    <div class="col-lg-12"><h3 class="entry-title"><a href="/2013/10/01/biosafety-level-4-research-coming-to-the-neidl">Biosafety Level 4 Research Coming to the NEIDL</a></h3></div>
</div>
<div class="row">
    <div class="col-lg-2">
        <div class="post-date">October 01, 2013</div>
    </div>
    <div class="col-lg-8">
        <p><p><img src="http://www.bu.edu/today/files/2012/07/t_12-5378-BUILDINGS-008.jpg" alt="neidl"></p>

<p>The National Emerging Infectious Diseases Laboratories (NEIDL) is a $200 million research facility built by Boston University. The National Institutes of Health (NIH) agreed to co-fund the project in 2006, but several court injunctions have kept most of its brand-new highly equipped laboratories ridiculously empty.</p>

<p>A recent <a href="http://www.bu.edu/neidl/files/2013/10/Judge-Saris-Ruling.pdf">federal ruling</a> by US District Court Chief Judge Patti Saris determined that the risks to the public associated with the NEIDL are &ldquo;extremely low, or beyond reasonably foreseeable&rdquo; and that its current location in Boston&rsquo;s South End neighborhood poses no greater threat than if it were located in a different suburban or rural area. One more state court lawsuit (which will be decided later this year) stands as the last major obstacle before <a href="https://en.wikipedia.org/wiki/Biosafety_level">BSL-3 and BSL-4 research</a> starts taking place at the NEIDL.</p>

<!--excerpt-->

<p>Maximum-security research facilities are the best place to pursue our ignorance about how these deadly pathogens work, and they provide the resources to eventually transform scary diseases into manageable annoyances. Unfortunately, a few vocal groups have pushed back against the construction of the NEIDL, claiming that its location would &ldquo;have a disproportionately adverse effect on minority and low-income population&rdquo;. This seems completely unfounded. If anyone was at risk, it would be the social contacts of a hypothetically infected lab worker, not the people living close to the facility.</p>

<p>Thirteen pathogens were chosen to be studied in the BSL-3 and BSL-4 facilities within the NEIDL, including the bacterium that causes anthrax, the 1918 flu virus, and the virus that causes SARS. The way I see it, the chances of catching the next swine flu variant coming from China far exceed those of getting infected from one of the pathogens being studied at the NEIDL.</p>

<p>The sooner research starts, the higher the chance that we will keep outbreaks from turning into epidemics.</p>
</p>
    </div>
</div>
<div class="clearfix"></div>
<hr class="list_separator" />

    <div class="row">
    <div class="col-lg-12"><h3 class="entry-title"><a href="/2013/09/29/what-happens-inside-of-a-mosquito-bite">What Happens Inside of a Mosquito Bite</a></h3></div>
</div>
<div class="row">
    <div class="col-lg-2">
        <div class="post-date">September 29, 2013</div>
    </div>
    <div class="col-lg-8">
        <p><p><img src="https://lh5.googleusercontent.com/-e2eDKxKdXhQ/Ukj7unNibYI/AAAAAAAAEgo/MjcF9ronqWI/w636-h824-no/mosquito-bite-picsbugman-bug-trivia-quiz-answer-8cjpkslo.jpg" alt="mosquito"></p>

<p>A mosquito bite is much more sophisticated than a needle prick. The mouthparts that penetrate the skin are surrounded by the labium (lb), which folds and stays outside, the maxillae help make the puncture, the labrum is the tube that sucks up the blood, and the hypopharynx sends saliva down. Once inside the skin, the labrum bends and probes until it finds a capillary (which can take up to two minutes).</p>

<!--excerpt-->

<p>Once the capillary is pierced, the mosquito sucks hardâ€”sometimes hard enough to rupture the capillary. The saliva contains anticoagulants, which keeps the blood flowing during the 4-minute-long feeding, and whatever parasites the mosquito was carrying. This video shows the biting process in all its gory details.</p>

<iframe width="640" height="480" src="//www.youtube.com/embed/MbXSPacvuak" frameborder="0" allowfullscreen></iframe>

<p><span class="caption">This video is freaky</span></p>

<p>The ice cubes are skin cells, the bendable hook is the mosquito straw, the red lines are the capillaries (you can almost see red blood cells). The images were captured by <a href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0050464">Valerie Choumet and her colleagues</a> at the Pasteur Institute in Paris by using a microscope to look through a flap of mouse skin. Check out the <a href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0050464#s5">supplementary section</a> for awesome pictures and videos.</p>

<p>I could talk about mosquito bites all day. Here are a couple of other <a href="http://www.nbcnews.com/id/3077327/ns/technology_and_science-science/t/how-mosquitoes-get-under-your-skin/#.UkjTO2RgaWY">awesome</a> <a href="http://phenomena.nationalgeographic.com/2013/08/06/heres-what-happens-inside-you-when-a-mosquito-bites/">articles</a>.</p>
</p>
    </div>
</div>
<div class="clearfix"></div>
<hr class="list_separator" />

    <div class="row">
    <div class="col-lg-12"><h3 class="entry-title"><a href="/2013/09/28/hunting-for-viruses-in-french-guiana">Hunting for Viruses in French Guiana</a></h3></div>
</div>
<div class="row">
    <div class="col-lg-2">
        <div class="post-date">September 28, 2013</div>
    </div>
    <div class="col-lg-8">
        <p><p><img src="https://lh3.googleusercontent.com/-itApYj3iVm4/UkdibZAnTbI/AAAAAAAAEdE/uG-5SBIlkao/w400-no/Oecomys+marmorae+22jan11+lag+bla+full.JPG" alt="oecomys">
<span class="caption">I don&rsquo;t know who that is, but it sure ain&rsquo;t me.</span></p>

<p>The International Division of the Pasteur Institute has generously awarded me a grant to carry out a 4-month research internship in its branch located in French Guiana. The goal of the project is to determine the viral biodiversity of multiple species of rodents living in different habitats in the region.</p>

<!--excerpt-->

<h2>Where I&rsquo;m going</h2>

<p><img src="https://lh6.googleusercontent.com/-hqdKNIO68e0/UkdLUP1ghVI/AAAAAAAAEcM/JvNUVAa-my8/w320-h350-no/french_guiana.gif" alt="French Guiana"></p>

<p>Although French Guiana borders Brazil and Suriname, it is not an independent country, but an overseas <em>region</em> of France. Therefore, it is part of the European Union, the official currency is the euro, and the license plates look European. The population is around 250,000 people, and half of them live in the capital, Cayenne. According to <a href="https://en.wikipedia.org/wiki/French_guiana">Wikipedia</a>, there is one road that crosses the country parallel to the coast, and the rest is Amazonian jungle.</p>

<p><img src="https://lh6.googleusercontent.com/-9e-Cn_NmC40/UkdMHH1VULI/AAAAAAAAEcg/qh_RpXeeGP0/w350-no/Screen+Shot+2013-09-28+at+5.37.02+PM.png" alt="FG up close"></p>

<p>The Pasteur Institute in French Guiana was created in 1940 and it employs around 100 scientists, distributed across 11 different labs. I will be working in the <a href="http://www.pasteur-cayenne.fr/spip/spip.php?article30&amp;lang=fr">Virus-Host Interactions lab</a>.</p>

<h2>What I&rsquo;m doing</h2>

<p>Whenever we urbanize or adapt a terrain for agriculture, we run the risk of disturbing the habitat of the species that live in it. This increases the chances that species that don&rsquo;t usually interact will come into contact with each other; and raises the possibility that these interactions will help pathogens cross the species barrier, thereby expanding their range. Many different species of rodents live in close proximity to humans, but we don&rsquo;t know much about the viruses they harbor. The little we know is that three rodent species are the likely reservoirs of some nasty viruses that killed several patients in French Guiana last year.</p>

<p>Metagenomics is a cool bioinformatics technique that can be used to determine all the genetic information that exists in a biological sample. This means that we can go capture rodents all over the country, take a few blood samples, process them and isolate what is likely to viral genetic material. Then, we sequence it and somebody gives me a hard drive full of As, Cs, Gs, and Tsâ€”which is when I come in, because I&rsquo;m not excited about getting bitten by virus-laden jungle rats.</p>

<p>My job is to determine what sequences are likely to belong to previously known viruses, to identify which ones may correspond to novel viruses (typically 50% of all the generated sequences are completely unknown!), and which ones may be potential human pathogens, and also, to compare the viral families in each sample and see if they correlate better by species or by habitat. Cool stuff.</p>
</p>
    </div>
</div>
<div class="clearfix"></div>
<hr class="list_separator" />

    <div class="row">
    <div class="col-lg-12"><h3 class="entry-title"><a href="/2013/09/27/teach-ignorance">Teach Ignorance!</a></h3></div>
</div>
<div class="row">
    <div class="col-lg-2">
        <div class="post-date">September 27, 2013</div>
    </div>
    <div class="col-lg-8">
        <p><p>Stuart Firestein is a neuroscience professor at Columbia University. For the past seven years, he has been teaching a course called <em>Ignorance</em>, where he invites scientists to talk to his students for a couple of hours about the state of their ignorance: what they didn&rsquo;t know a few years ago and know now, what they would still like to know, why it is important to know it, and how they plan to find out. He&rsquo;s funny and clever. His TED talk just came out:</p>

<iframe src="http://embed.ted.com/talks/stuart_firestein_the_pursuit_of_ignorance.html" width="640" height="360" frameborder="0" scrolling="no" webkitAllowFullScreen mozallowfullscreen allowFullScreen></iframe>

<!--excerpt-->

<h2>Insightful ignorance drives science</h2>

<p>Knowing lots of facts doesn&rsquo;t make you a scientist, you also need to ask good questions. Scientists use facts to ask better questions, insightful questions, questions that probe at the ignorance and expand the frontier of what&rsquo;s known. Strangely, science is not taught as this exciting, endless pursuit of questions. Rather, it is delivered as lecture-sized bundles of facts that must be memorized to pass the test and have little relevance afterwards. Facts should not be simply accumulated, they should be used to build the next round of questions.</p>

<p>You can look up what we know, but finding out what we don&rsquo;t know requires having original thoughts. The current educational system doesn&rsquo;t know how to deal with this requirement. I love the proposal that Stuart shows at the end of his talk about getting rid of tests that ask for the right answer (which Google knows), and replacing them with tests that give the answer, and ask what is the next question (which Google doesn&rsquo;t know).</p>

<p>Instead of drowning students in pointless facts, teachers should spend most of their time lighting their curiosity. Get them interested enough, and they&rsquo;ll find out the answer to their questions by themselves.</p>

<p>If you liked his talk, be sure to listen to his <a href="http://www.twiv.tv/2012/12/29/twiv-special-ignorance-with-stuart-firestein/">interview</a> on the podcast This Week in Virology. He also has a <a href="http://www.amazon.com/Ignorance-How-Drives-Science-ebook/dp/B007M7HXMC/ref=sr_1_1?s=books&amp;ie=UTF8&amp;qid=1380324600&amp;sr=1-1&amp;keywords=ignorance">book</a>.</p>
</p>
    </div>
</div>
<div class="clearfix"></div>
<hr class="list_separator" />

    <div class="row">
    <div class="col-lg-12"><h3 class="entry-title"><a href="/2013/09/25/bioinformatics-at-the-bench">Bioinformatics at the Bench</a></h3></div>
</div>
<div class="row">
    <div class="col-lg-2">
        <div class="post-date">September 25, 2013</div>
    </div>
    <div class="col-lg-8">
        <p><p>Most bioinformaticians don&rsquo;t like to do experiments. We love the biological stories, but give us a pipette and we will quickly groan. I have spent the past two years in a virology lab, mainly working on computational problems, and eternally debating if I should plunge into the pool of experimental biology. Up until now, I have only dipped my toes in a few basic experiments, but I have recently decided to stop working on purely computational projects and transition to ones that require experimental skills. Here are my reasons.</p>

<!--excerpt-->

<h2>I want to learn what&rsquo;s possible</h2>

<p>Computationally, anything I can imagine is possible. Experimentally, I only have a vague notion about what the space of possibility looks like. One of the reasons why I joined an experimental lab was to fill this gap, but I have finally realized that this doesn&rsquo;t happen by osmosis. I will never understand what techniques are commonly used and which ones are cutting-edge without studying the literature. Also, no amount of reading will teach me how to troubleshoot experiments that don&rsquo;t work; I need to interact with people that have spent years acquiring the wisdom.</p>

<h2>I want to work on more interesting problems</h2>

<p>Collaborations between bioinformaticians and biologists typically work like this: 1) biologist approaches bioinformatician with a ton of data and a few questions, 2) bioinformatician spends days analyzing the data, 3) bioinformatician shares findings and waits for biologist to experimentally validate them. This rarely leads to ground-breaking results because computational analyses usually underestimate the biological complexity. Bioinformatics can be used as a hypothesis generation tool, but I find that relegating the experimental component to the end is a mistake. The ability to design and perform any experiment, and the ability to process and analyze any kind of dataset are two ridiculously useful techniques on their own, but when applied together, their usefulness compounds.</p>
</p>
    </div>
</div>
<div class="clearfix"></div>
<hr class="list_separator" />

    <div class="row">
    <div class="col-lg-12"><h3 class="entry-title"><a href="/2013/09/24/why-10-of-the-population-hates-cilantro-and-the-rest-doesnt-know-any-better">Why 10% of the Population Hates Cilantro and the Rest Doesn't Know Any Better</a></h3></div>
</div>
<div class="row">
    <div class="col-lg-2">
        <div class="post-date">September 24, 2013</div>
    </div>
    <div class="col-lg-8">
        <p><blockquote>
<pre>
"No cilantro, please"
Yet I know when it arrives
It will be present</pre>

<p>Haiku by <a href="http://www.ihatecilantro.com/member.php?user=Krakrs">Krakrs</a>, a member of <a href="http://ihatecilantro.com">ihatecilantro.com</a></p>
</blockquote>

<p>The first time I tried cilantro I didn&rsquo;t realize it; I just thought somebody had emptied a bottle of Old Spice on my pizza in an attempt to poison me. Cilantro tastes like soap to approximately 10% of the people who have had their genotype analyzed by <a href="http://www.biomedcentral.com/content/pdf/2044-7248-1-22.pdf">23andMe</a>. The currently accepted explanation is that those of us who passionately despise cilantro were born with a genetic variant known as a <strong>single-nucleotide polymorphism</strong> (or <a href="http://en.wikipedia.org/wiki/Single-nucleotide_polymorphism">SNP</a>, pronounced &lsquo;snip&rsquo;).</p>

<!--excerpt-->

<p>The genome has 3 billion nucleotides (the building blocks, known as A, C, G and T), and 10 million of them are thought to be SNPs. That means that a significant percentage of the population has one letter in a specific location (an A, for example) and everyone else has a different letter at that location. The cilantro SNP is called <a href="http://snpedia.com/index.php/Rs72921001">rs72921001</a>, and apparently, its genomic location lays close to a cluster of olfactory receptor genes that includes OR6A2, the gene most likely to be alerting our brain about the presence of cilantro.</p>

<p><img src="https://lh4.googleusercontent.com/-P5G_ZAWbiyg/UkJfWn69PlI/AAAAAAAAEYs/DhvCPLBNhZw/w200-no/Screen+Shot+2013-09-24+at+11.57.10+PM.png" alt="Olfactory receptor">
<span class="caption">A cartoon representation of the molecular structure of an olfactory receptor. You can think of the red compound snuggling between the helices as the tiniest bit of cilantro.</span></p>

<p>When molecules enter the nose, they come into contact with the olfactory sensory neurons located in the upper region of the nasal cavity. Each of these cells expresses only one odorant receptor, and when the right chemical binds to the right receptor, an electrical signal is sent upwardsâ€”through the boneâ€”to the olfactory bulb, where it is further relayed by other cells until it reaches the olfactory cortex region of the brain. The OR6A2 receptor gets activated by binding to one of the several <a href="https://en.wikipedia.org/wiki/Aldehyde">aldehydes</a> contained in cilantro.</p>

<p><img src="https://lh6.googleusercontent.com/-_4YSK2rRkKQ/UkJfV9EdeNI/AAAAAAAAEYk/hRNqVOwTHAE/w500-no/Screen+Shot+2013-09-24+at+11.09.05+PM.png" alt="How smell works">
<span class="caption">A figure from <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1905909/pdf/7401029.pdf">The scent of life</a>, detailing how smell works.</span></p>

<p>The letter you have in the location of the rs72921001 SNP is thought to influence how the OR6A2 receptor perceives the presence of cilantro. If you are among the 10% of humans who have a C instead of an A in that location, when confronted with cilantro, instead of the misleading &ldquo;Gee, I smell cilantro, lalala, happiness&rdquo; your brain will warn you of the real danger you face: &ldquo;I smell the disgusting herb that poisons everything good and loveworthy in this world. Run for your life!&rdquo;</p>

<p>The science of smell is fascinatingly complex and not entirely understood, but the current hypothesis is that having a C instead of an A in a specific location close to the OR6A2 gene, slightly alters the shape of the receptor molecule encoded by that gene, and allows cilantro molecules to activate receptors that in the rest of the population are only activated by soap/bleach/death molecules.</p>
</p>
    </div>
</div>
<div class="clearfix"></div>
<hr class="list_separator" />

    <div class="row">
    <div class="col-lg-12"><h3 class="entry-title"><a href="/2013/09/21/measuring-the-effects-of-interferon-stimulated-genes-on-viral-replication">Measuring the Effects of Interferon Stimulated Genes on Viral Replication</a></h3></div>
</div>
<div class="row">
    <div class="col-lg-2">
        <div class="post-date">September 21, 2013</div>
    </div>
    <div class="col-lg-8">
        <p><p>Whenever I read a paper that describes an experimental technique that I&rsquo;m not familiar with (which is often, since the lab bench is not where bioinformatic students spend most their time), I compensate by doing background research. Lately, I have been reading a lot about the antiviral effects of interferon-stimulated genes (ISGs), and since <a href="http://www.utsouthwestern.edu/labs/schoggins/">John Schoggins</a> is giving a seminar in my university on Monday, I thought I would post about a clever technique that he used in his <a href="http://www.nature.com/nature/journal/v472/n7344/abs/nature09907.html">Nature paper</a> to quantify the impact of ISGs on viral replication.</p>

<!--excerpt-->

<p>Interferon-stimulated genes are known to inhibit viral replication, but this effect hasn&rsquo;t been quantified for all of them. The goal of the Schoggins paper was to determine how individual ISGs change the ability of different viruses to replicate.</p>

<p>The authors used a <strong>bicistronic lentiviral vector</strong>, which is molecular biology jargon for a piece of DNA that contains two genes that get inserted into the genome of a cell by using a lentivirus virion. They are also known as plasmids or constructs. Besides the two genes, the foreign DNA also contains a bunch of other interesting molecular tools assembled in Frankenstein-like fashion:</p>

<p><img src="https://lh3.googleusercontent.com/-f1HDGbsiX9o/Uj4b8UQ4PfI/AAAAAAAAEU8/VhtcGWcfVoU/w500-no/Screen+Shot+2013-09-21+at+6.20.32+PM.png" alt="DNA construct"></p>

<ul>
<li><p><strong>LTR</strong> stands for long terminal repeat. It is a sequence of nucleotides repeated hundreds of times. Viruses (especially retroviruses, like HIV) use them to insert their genetic material into the genome of their target cell. In this case, instead of inserting a viral genome, they help integrate the plasmid into the genome of a cell.</p></li>
<li><p><strong>CMV</strong> is one of the enhancer sequences of the human cytomegalovirus genome. It is another viral tool to increase the amount of gene transcription.</p></li>
<li><p><strong>IVS-beta</strong> is the second intron in the rabbit beta-globin gene (one of the chains that makes up hemoglobin). Adding it to the  construct also increases the transcription of the plasmid genes. I haven&rsquo;t been able to really understand why, but I think it has to do with how it interacts with the splicing machinery.</p></li>
<li><p><strong>ISG</strong> is the sequence for one of the 300+ known interferon-stimulated genes. Inserting it after the two previous elements ensures that it will be highly transcribed.</p></li>
<li><p><strong>EMCV IRES</strong> is the internal ribosomal entry site of the encephalomyocarditis virus. This is another page taken from the virus book: it is a nucleotide sequence that allows ribosomes to bind to the middle of the messenger RNA molecule, instead of the more traditional 5&rsquo; end. Viruses use it to synthesize their proteins when the host translational machinery is not supposed to be working. In the plasmid above, it ensures that the RFP gene is translated in a proportional amount to the ISG gene.</p></li>
<li><p><strong>RFP</strong> stands for red fluorescent protein. Using it as the second protein of the plasmid makes it easy to identify which cells have incorporated the plasmid into their genome, and are therefore, also expressing the ISG of interest.</p></li>
</ul>

<p>The viruses used in the experiment were modified so they would express GFP (green fluorescent protein). That way, the amount of viral replication can be quantified by measuring the amount of red fluorescence (from cells that incorporated the plasmid) and green fluorescence (from cells that were also infected with the virus). Each ISG construct has a specific ability to reduce viral replication. Surprisingly, a number of ISGs actually <em>enhance</em> viral replication (the circles that are above 100% in the graph below), although the reasons for this are not yet clear.</p>

<p><img src="https://lh4.googleusercontent.com/-QijjpmdVg1c/Uj4xAaxoIxI/AAAAAAAAEVY/4aezYjRCcS8/w431-h456-no/Screen+Shot+2013-09-21+at+7.47.28+PM.png" alt="isgs"></p>

<p>Clever molecular techniques like this one are the best tool that we have to understand what is really going on inside our cells. By studying them, I hope to learn enough to eventually design my own useful experiments.</p>
</p>
    </div>
</div>
<div class="clearfix"></div>
<hr class="list_separator" />

    <div class="row">
    <div class="col-lg-12"><h3 class="entry-title"><a href="/2013/09/20/my-phd-research-project-diagnosing-viral-infections">My PhD Research Project: Diagnosing Viral Infections</a></h3></div>
</div>
<div class="row">
    <div class="col-lg-2">
        <div class="post-date">September 20, 2013</div>
    </div>
    <div class="col-lg-8">
        <p><p>I gave a presentation today in the Microbiology departmental retreat at Boston University, so I though I&rsquo;d share it.</p>

<p><img src="https://lh5.googleusercontent.com/-nzVbaorZHoY/UjzzWilVUuI/AAAAAAAAEUA/5--h0EgJKbU/w500-no/lassa+title.png" alt="presentation"></p>

<p>The goal of the research project I have been working on for the past two years, as part of my Bioinformatics PhD at Boston University, is to use blood samples from infected patients to determine which virus is causing the infection.</p>

<p>Traditional diagnostic methods, like detection by <a href="https://en.wikipedia.org/wiki/ELISA">ELISA</a>, are only effective after the virus has had enough time to replicate in the blood of the patient. However, for some viral infections, when this happens it is already too late for the treatment to be effective.</p>

<p>The approach that we have taken in my lab is to measure the transcriptional changes that take place in the peripheral blood cells, and identify patterns of expression that are unique for each type of infection. This indirect way of viral detection has the potential to reduce diagnosis time by several days, since it is known that these transcriptional changes precede the appearance of virus particles in the blood, and we believe that they are specific enough to discriminate between viruses.</p>

<!--excerpt-->

<p>To test our hypothesis, we sequenced the RNA from <a href="https://en.wikipedia.org/wiki/PBMC">PBMCs</a> that were extracted from two groups of monkeys; one infected with Lassa virus, and the other with Marburg virus.</p>

<p>After comparing the expression of the samples extracted 3 days post-infection with those taken before infection, we identified multiple genes that showed strong changes in expression. Most of these genes play important roles in innate immunity pathways, so we expected them to be highly expressed in both infections.</p>

<p>What we found most surprising was the number of genes that showed different patterns of expression depending on what virus was causing the infection. Using only the expression patterns of a handful of genes, we were able to correctly classify unknown samples into three groups: uninfected, infected with Lassa virus, and infected with Marburg virus.</p>

<p>Our hope is that these results will be integrated in a diagnostic kit to easily identify Lassa and Marburg infected patients in areas where these diseases are common, but there is still a long way to go.</p>

<p>Here is the presentation I gave today.</p>

<iframe src="http://www.slideshare.net/slideshow/embed_code/26398948" width="600" height="500" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC;border-width:1px 1px 0;margin-bottom:5px" allowfullscreen webkitallowfullscreen mozallowfullscreen> </iframe>
</p>
    </div>
</div>
<div class="clearfix"></div>
<hr class="list_separator" />

    <div class="row">
    <div class="col-lg-12"><h3 class="entry-title"><a href="/2013/09/12/turning-bacteria-into-genetic-circuits-without-them-noticing">Turning Bacteria into Genetic Circuits (Without Them Noticing)</a></h3></div>
</div>
<div class="row">
    <div class="col-lg-2">
        <div class="post-date">September 12, 2013</div>
    </div>
    <div class="col-lg-8">
        <p><p>Lately I have been summarizing every paper I read on a single piece of paper. Before I started doing that I would invest a lot of time reading a paper, and two months later it would completely disappear from my brain. The only thing I would remember is if I had found it interesting or not (not the most important information when trying to compile a bibliography). Limiting myself to a single piece of paper ensures that I only record what is important, and the fact that the notes are handwritten lets me <a href="http://reasoniamhere.com/2013/09/09/good-presentations-begin-on-paper/">triple underscore and use all the curvy arrows</a> I need to make  important ideas stand out. My two-month recall is now pretty decent.</p>

<p>I&rsquo;m going to a seminar tomorrow presented by Tim Lu, one of the authors of a Nature Biotechnology paper titled <a href="http://www.rle.mit.edu/sbg/wp-content/uploads/2013/03/synthetic-circuits-integrating-logic-and-memory-in-living-cells.pdf">Synthetic circuits integrating logic and memory in living cells</a>. I decided to read the paper and write up a quick summary.</p>

<p><img src="/assets/synthetic_circuits.jpg" alt="Synthetic Circuits"></p>

<!--excerpt-->

<p>The paper describes a new technique to arrange the genome of a bacterium (<em>E. coli</em>) and turn it into the biological equivalent of an electronic circuit. This approach is much more scalable than previous techniques because using it to build complex circuits like an XOR gate doesn&rsquo;t require joining universal gates together, which means that instead of assembling this madness:</p>

<p><img src="http://upload.wikimedia.org/wikipedia/commons/thumb/4/4e/NOR_from_NAND.svg/280px-NOR_from_NAND.svg.png" alt="An XOR gate using NANDs"></p>

<p>you can assemble the XOR gate directly:</p>

<p><img src="http://upload.wikimedia.org/wikipedia/commons/thumb/0/01/XOR_ANSI.svg/100px-XOR_ANSI.svg.png" alt="An XOR gate"></p>

<p>(By the way, a gate is just a way to convert two inputs into one output. Inputs and outputs can be either 1s or 0s. The type of gate (AND, OR, XOR, etc.) describes the rule that is used to make this conversion. For example, check out the table rule for the <a href="http://en.wikipedia.org/wiki/AND_gate">AND gate</a>)</p>

<p>Stinginess is important because you can use as many components as you can fit in a circuit boards, but cells are much more puritan about the number of resources they allow.</p>

<p>The paper&rsquo;s magic sauce is in the use of <strong>inducible recombinases</strong>â€”recombinases that become active when their inducer (a riboregulator) is added to the media. An activated recombinase can bind specific regions of DNA and flip their orientation, which in turn switches on or off the expression of their downstream gene.</p>

<p><img src="/assets/and_gate.png" alt="An AND gate"></p>

<p>For example, the blue brackets in the figure above represent the DNA binding site of a recombinase that, once activated by its inducer (input A), flips that region of DNA, making the promoter (black arrow) accessible to the RNA polymerase. But the RNA polymerase has nothing to transcribe unless the region between the red triangles is also flipped by a different recombinase. When this recombinase gets activated by its inducer (input B), the Green Fluorescent Protein gene can be transcribed and the cell starts making enough GFPs to turn the bacterium into a swimming Christmas tree. As the barplot shows, when both inputs are not present, the amount of GFP that is made is minimal (no biological circuit will ever be perfect).</p>

<p>(Just to clarify, GFP is an example, any gene can be integrated into this type of circuit. It&rsquo;s just that a glowing bacterium is easier to detect than one that expresses the blond-hair gene)</p>

<p>That is just one possibility out of the 16 that appear in Figure 2 (<a href="http://www.rle.mit.edu/sbg/wp-content/uploads/2013/03/synthetic-circuits-integrating-logic-and-memory-in-living-cells.pdf">check out</a> the especially clever technique used to build the XOR and XNOR gates).</p>

<p>Because the recombinases modify the information that is stored in the bacterial chromosome, after a bacterium divides, the two daughter cells have the same genetic information, which means that next generations of bacteria will keep synthesizing GFP for centuries (days, in bacterium-years). DNA is what Nature uses as a hard drive, which means that coming up with clever ways to store information in it will bring us ever closer to genetically engineering the bionic pets we have all been dreaming about.</p>

<p>The <a href="(http://www.rle.mit.edu/sbg/wp-content/uploads/2013/03/synthetic-circuits-integrating-logic-and-memory-in-living-cells.pdf)">paper</a> is blissfully short, and worth a read if you think synthetic biology is cool.</p>
</p>
    </div>
</div>
<div class="clearfix"></div>
<hr class="list_separator" />



    


<div class="pagination">
      <ul>
      
        <li class="prev disabled"><a>&larr; Previous</a></li>
      
        <li><a href="/archive.html">Article Index</a></li>
      
        <li class="next disabled"><a>Next &rarr;</a>
      
      </ul>
    </div>
      </div>

      <footer>
        <p class="copyright">&copy; 2015 Nacho Caballero</p>
        <p>
          <a href="http://feeds.feedburner.com/TheReasonIAmHere">RSS</a> â€¢
          <a href="https://twitter.com/nachocaballero">Twitter</a> â€¢
          <a href="http://reasoniamhere.com/about.html">About</a>
        </p>
      </footer>

    </div>
    <script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-42664204-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>
  </body>
</html>

